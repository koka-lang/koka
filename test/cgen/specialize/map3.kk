fun map-poly(xs : list<a>, f : a -> e b) : e list<b> {
  match (xs) {
    Nil -> Nil
    Cons(x, xx) -> Cons(x.f, xx.map-poly(f)) 
  }
}

public fun test(xs : list<int>, y : int) {
  xs.map-poly(fn(x){ x + y })   // free variable
}

public fun main() {
  list(1,10).test(5).show.println
}
