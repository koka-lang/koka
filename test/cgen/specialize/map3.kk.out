[6,7,8,9,10,11,12,13,14,15]
 
cgen/specialize/map3/.lift000-test: (y : int, xs0 : list<int>) -> list<int>
cgen/specialize/map3/.lift000-main: (xs : list<int>) -> list<int>
cgen/specialize/map3/.mlift000-map-poly: forall<a,e> (a, list<a>) -> e list<a>
cgen/specialize/map3/.mlift000-map-poly: forall<a,b,e> (f : (a) -> e b, xx : list<a>, b) -> e list<b>
cgen/specialize/map3/main: () -> console ()
cgen/specialize/map3/map-poly: forall<a,b,e> (xs : list<a>, f : (a) -> e b) -> e list<b>
cgen/specialize/map3/test: (xs : list<int>, y : int) -> list<int>